src/py21cmfast/inputs.py

```diff
@@ -0,0 +1,614 @@
+
+    To see default values for each parameter, use ``CosmoParams._defaults_``.
+    All parameters passed in the constructor are also saved as instance attributes which should
+    be considered read-only. This is true of all input-parameter classes.
+
+    Parameters
+    ----------
+    SIGMA_8 : float, optional
+        RMS mass variance (power spectrum normalisation).
+    hlittle : float, optional
+        The hubble parameter, H_0/100.
+    OMm : float, optional
+        Omega matter.
+    OMb : float, optional
+        Omega baryon, the baryon component.
+    POWER_INDEX : float, optional
+        Spectral index of the power spectrum.
+    """
+
+    _ffi = ffi
+
+    _defaults_ = {
+        "SIGMA_8": 0.82,
+        "hlittle": Planck15.h,
+        "OMm": Planck15.Om0,
+        "OMb": Planck15.Ob0,
+        "POWER_INDEX": 0.97,
+    }
+
+    @property
+    def OMl(self):
+        """Omega lambda, dark energy density."""
+        return 1 - self.OMm
+
+    @property
+    def cosmo(self):
+        """Return an astropy cosmology object for this cosmology."""
+        return Planck15.clone(H0=self.hlittle * 100, Om0=self.OMm, Ob0=self.OMb)
+
+
+class UserParams(StructWithDefaults):
+    """
+    Structure containing user parameters (with defaults).
+
+    To see default values for each parameter, use ``UserParams._defaults_``.
+    All parameters passed in the constructor are also saved as instance attributes which should
+    be considered read-only. This is true of all input-parameter classes.
+
+    Parameters
+    ----------
+    HII_DIM : int, optional
+        Number of cells for the low-res box. Default 50.
+    DIM : int,optional
+        Number of cells for the high-res box (sampling ICs) along a principal axis. To avoid
+        sampling issues, DIM should be at least 3 or 4 times HII_DIM, and an integer multiple.
+        By default, it is set to 4*HII_DIM.
+    BOX_LEN : float, optional
+        Length of the box, in Mpc. Default 150.
+    HMF: int or str, optional
+        Determines which halo mass function to be used for the normalisation of the
+        collapsed fraction (default Sheth-Tormen). If string should be one of the
+        following codes:
+        0: PS (Press-Schechter)
+        1: ST (Sheth-Tormen)
+        2: Watson (Watson FOF)
+        3: Watson-z (Watson FOF-z)
+    USE_RELATIVE_VELOCITIES: int, optional
+        Flag to decide whether to use relative velocities, default True.
+        If True, POWER_SPECTRUM is automatically set to 5. Default False.
+    POWER_SPECTRUM: int or str, optional
+        Determines which power spectrum to use, default EH (unless `USE_RELATIVE_VELOCITIES`
+        is True). If string, use the following codes:
+        0: EH
+        1: BBKS
+        2: EFSTATHIOU
+        3: PEEBLES
+        4: WHITE
+        5: CLASS (single output)
+    """
+
+    _ffi = ffi
+
+    _defaults_ = {
+        "BOX_LEN": 150.0,
+        "DIM": None,
+        "HII_DIM": 50,
```

Comment:
This is a discussion for another time, but I've always found it dubious that the default BOX_LEN is 150 Mpc, since that can be too small to capture all the matter fluctuations (k>0.05 Mpc^-1).

Reply:
I think that's a fair point. The default is supposed to be really quite small, so that if you fire up a session and just run with defaults, it doesn't take a bazillion years. But I'll pull up a separate issue for it.